sale_stock_analytic/models/stock_rule.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+
+    _inherit = "stock.rule"
+
+    def _get_stock_move_values(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_id,
+        name,
+        origin,
+        company_id,
+        values,
+    ):
+        move_values = super(StockRule, self)._get_stock_move_values(
+            product_id,
+            product_qty,
+            product_uom,
+            location_id,
+            name,
+            origin,
+            company_id,
+            values,
+        )
+        sol_id = move_values.get("sale_line_id", False)
+        if sol_id:
+            sol_model = self.env["sale.order.line"]
+            sol = sol_model.browse(sol_id)
```

Comment:
We should consider the case where the analytic account is at the sale order level. In that case, we have to take the sale order analytic account on every line of the stock move

Reply:
I did a PR on your branch
https://github.com/DynAppsNV/account-analytic/pull/1

Reply:
I don't think we have to do this. Passing `analytic_distribution` to stock move is enough.